storage/storage_interface.ml

```diff
@@ -22,28 +78,54 @@ let default_sockets_dir = "/var/lib/xcp"
 let default_path = ref (Filename.concat default_sockets_dir service_name)
 
 let set_sockets_dir x =
-        default_path := Filename.concat x service_name
+  default_path := Filename.concat x service_name
 
 let uri () = "file:" ^ !default_path
 
+let rpc_of t x = Rpcmarshal.marshal t.Rpc.Types.ty x
+let of_rpc t x = match Rpcmarshal.unmarshal t.Rpc.Types.ty x with | Ok y -> y | Error (`Msg m) -> failwith (Printf.sprintf "Error unmarshalling type %s: %s" t.Rpc.Types.name m)
 
 (** Primary key identifying the SR *)
-type sr = string
+module Sr : sig
+  type t
+  val typ_of : t Rpc.Types.typ
+  val t : t Rpc.Types.def
+  val string_of : t -> string
+  val of_string : string -> t
+end = struct
+  type t = string [@@deriving rpcty]
+  let string_of x = x
+  let of_string x = x
+end
+
+type sr = Sr.t
 
 (** Primary key identifying a VDI within an SR *)
-type vdi = string
+module Vdi : sig
+  type t
+  val typ_of : t Rpc.Types.typ
+  val t : t Rpc.Types.def
+  val string_of : t -> string
+  val of_string : string -> t
+end = struct
```

Comment:
Should we have a common signature for these? (Maybe call it WrappedString)

Reply:
That makes sense to me.

Reply:
I'll update the PR.